From d8a0476ff1d36f7e0a92129c687b69f38830abb3 Mon Sep 17 00:00:00 2001
From: Venkatesh Yadav Abbarapu <vabbarap@xilinx.com>
Date: Mon, 11 Dec 2017 09:52:24 +0530
Subject: [PATCH] Update the logic to check if cpu is present in the design or
 not

Some of the designs doesn't have all the cpus, so add the logic to
fix this.

Signed-off-by: Venkatesh Yadav Abbarapu <vabbarap@xilinx.com>
Acked-by: Michal Simek <michals@xilinx.com>
---
 device_tree/data/device_tree.tcl | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/device_tree/data/device_tree.tcl b/device_tree/data/device_tree.tcl
index dd3928b..a838141 100644
--- a/device_tree/data/device_tree.tcl
+++ b/device_tree/data/device_tree.tcl
@@ -377,6 +377,11 @@ proc update_cpu_node {os_handle} {
     set proc_instance 0
     for {set i 0} {$i < $total_cores} {incr i} {
         set proc_name [lindex [get_cells -hier -filter {IP_TYPE==PROCESSOR}] $i]
+        if {[llength $proc_name] == 0} {
+            set cpu_node [add_or_get_dt_node -n "cpus" -d ${default_dts} -p ${system_root_node}]
+            hsi::utils::add_new_dts_param "${cpu_node}" "/delete-node/ cpu@$i" "" boolean
+            continue
+        }
 	if {[string match -nocase [get_property IP_NAME [get_cells -hier $proc_name]] "microblaze"]} {
 		return
 	}
-- 
2.7.4

